libraries/fabric-shim/lib/cmds/serverCommand.js

```diff
@@ -45,6 +53,38 @@ exports.getArgs = function (yargs) {
         argv[name] = yargs.argv[name];
     }
 
+    // Additional checks for TLS and translate to tlsProps
+    if (typeof argv['chaincode-tls-key-file'] === 'string' ||
```

Comment:
It's possible if I recall to set conditions such as these via yargs configuration.  

Can you clarify the thinking of checking the types? They are set for strings in the yargs config.

Reply:
Thank you for the review. The intention of checking types was just a null check.
But since it might not be very obvious, I have changed these conditions to just `argv['option']`.

The condition here is:
- If any of these four options are set, they must meet the following two conditions:
  - Exactly one of the two options "chaincode-tls-key-file" and "chaincode-tls-key-path" should be set
  - Exactly one of the two other options "chaincode-tls-cert-file" and "chaincode-tls-cert-path" should be set.

I could not find yargs configuration to express this condition exactly...so I have changed the code to use `.conflicts` for mutual exclusion of "-file" and "-path", and to use `check` for a custom logic.

Reply:
great thanks.. that's along the lines of what I was thinking.